### PR TITLE
Fix #624 - Add analytics to the service alerts button

### LINKED ISSUE #624: Add analytics to the "service alerts button"
**Summary:** 

We would like to report how many times a service alert button is clicked. Therefore, we need to add analytics when a service alert button is clicked. We should also include the agency id to this report.

### PR DESCRIPTION
**Pull request by cagryInside**
Following analytics report was added when a `service alerts button` is clicked:

 Event Category  | Event Action | Event Label |
| ------------- | ------------- | ------------- |
| ui_action  | service_alerts  | Clicked Service Alerts from: PSTA |

Screenshot from GA console:

untitled 2

cc'd @barbeau

### CODE REVIEW

**Comment by barbeau** [comment]
Thanks @cagryInside!  

@aaronbrethorst Would you be able to get this out in the next release?

**Comment by bbodenmiller** [comment]
I don't think you need to specify the region as I believe you can determine that by cross referencing the data.

**Comment by cagryInside** [comment]
> I don't think you need to specify the region as I believe you can determine that by cross referencing the data.

@bbodenmiller To clarify, it is not region name, it is `agency_id`. Since, a region can contain multiple agencies, this helps us to differentiate service alerts from different agencies in the same region.

File: OBAKit/Models/unmanaged/OBAReferencesV2.h [hunk 1]
```diff
@@ -17,6 +17,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (void) addAgency:(OBAAgencyV2*)agency;
 - (OBAAgencyV2*) getAgencyForId:(NSString*)agencyId;
+- (NSMutableDictionary*) getAllAgencies;
```
Comment (aaronbrethorst) [inline OBAKit/Models/unmanaged/OBAReferencesV2.h]: This class shouldn't publicly vend a mutable dictionary. I'd like to see the method signature changed to `NSDictionary`

File: OBAKit/Models/unmanaged/OBAReferencesV2.m [hunk 2]
```diff
@@ -27,7 +27,11 @@ - (void) addAgency:(OBAAgencyV2*)agency {
 
 - (OBAAgencyV2*) getAgencyForId:(NSString*)agencyId {
     return _agencies[agencyId];
-}    
+}
+
+- (NSMutableDictionary*) getAllAgencies {
+    return _agencies;
```
Comment (aaronbrethorst) [inline OBAKit/Models/unmanaged/OBAReferencesV2.m]: `return [NSDictionary dictionaryWithDictionary:_agencies];`

File: ui/situations/OBASituationsViewController.m [hunk 3]
```diff
@@ -93,6 +94,17 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
...
         [self.navigationController pushViewController:vc animated:YES];
+        [self reportAnalytics:situation];
+    }
+}
+
+- (void) reportAnalytics:(OBASituationV2 *) situation {
+    
+    NSMutableDictionary *agencies = [[situation references] getAllAgencies];
+    for(id key in agencies) {
+         OBAAgencyV2 * agency = [agencies objectForKey:key];
```
Comment (aaronbrethorst) [inline ui/situations/OBASituationsViewController.m]: please use object literal syntax: `agencies[key]`

**Comment by aaronbrethorst** [comment]
please make the changes described above and squash the resulting commits, and then i'll be happy to merge. thank you!

**Comment by cagryInside** [comment]
> please make the changes described above and squash the resulting commits, and then i'll be happy to merge. thank you!

@aaronbrethorst thanks for your comments. Latest commit fixes the in-line comments.

**Comment by aaronbrethorst** [comment]
looks good!

**Pull request merged**